### PR TITLE
Make debug output optional via an env variable

### LINKED ISSUE #6: Make debug output optional
Lots and lots of debug-level output is emitted to standard error.

It would be nice to make this debug output optional, perhaps via a command-line parameter (--debug?) or environment variable.

### PR DESCRIPTION
**Pull request by snapdgn**
Emits debug output to stderr via `DEBUG_OUT` env variable.
ex: `env DEBUG_OUT=1 make json 2>stderr.txt ` (fish shell).
Let me know if this seems alright.
Should i also include a --debug flag ?
Resolves #6

### CODE REVIEW

File: src/sail_json_backend/json.ml [hunk 1]
```diff
@@ -71,6 +71,11 @@ open Ast
 open Ast_defs
 open Ast_util
 
+let debug_enabled =
+  match Sys.getenv_opt "DEBUG_OUT" with
```
Comment (ThinkOpenly) [inline src/sail_json_backend/json.ml]: "DEBUG_OUT" is fairly generic, and I worry about colliding with other uses of a variable with the same name.

Let's add some namespace to the environment variable, calling it "SAIL_DEBUG", perhaps?

File: src/sail_json_backend/json.ml [hunk 2]
```diff
@@ -91,60 +96,60 @@ let string_of_arg = function
 
 let rec string_list_of_mpat x = match x with
   | MP_aux (MP_lit ( l ), _) ->
-      prerr_endline ("MP_lit " ^ string_of_lit l);
+      if debug_enabled then prerr_endline ("MP_lit " ^ string_of_lit l);
```
Comment (ThinkOpenly) [inline src/sail_json_backend/json.ml]: This logic is repeated with every instance. I suggest moving it into a new function, say `debug_print`?

**Review by ThinkOpenly — CHANGES_REQUESTED** [review]
Thanks for your efforts here. I have a couple of minor suggestions.

File: src/sail_json_backend/json.ml (hunk 2 above)
Comment (snapdgn) [reply src/sail_json_backend/json.ml]: sure, will do.

**Review by ThinkOpenly — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**